thoth/storages/graph/postgres.py

```diff
@@ -1880,6 +1880,78 @@ def get_analyzed_error_python_package_versions_count_all(
 
     # Unanalyzed Python Packages
 
+    def _construct_si_unanalyzed_python_package_versions_query(
+        self, session: Session, os_name: str = None, os_version: str = None, python_version: str = None, index_url: str = None
+    ) -> Query:
+        """Construct query for packages analyzed by solver, but unanalyzed by SI."""
+        index_url = GraphDatabase.normalize_python_index_url(index_url)
+        query = session.query(PythonPackageVersion).filter(
+            PythonPackageVersion.package_version.isnot(None),
+            PythonPackageIndex.url.isnot(None),
+            PythonPackageIndex.enabled.is_(True),
+        )
+
+        if index_url is not None:
+            query = query.filter(PythonPackageIndex.url == index_url)
+
+        # We find all rows that are same in PythonPackageVersion and SIAggregated table.
+        conditions = [PythonPackageVersion.entity_id == SIAggregated.id]
```

Comment:
SIAggreated is not linked to PPV table, this condition would lead to wrong results: https://github.com/thoth-station/storages/blob/4a74226545c0a36dbca7ec248058b71905c94c6c/thoth/storages/graph/models.py#L1550
```
class SIAggregated(Base, BaseExtension):
    """A relation representing a Python package version entity analyzed by a si_aggregated_run."""

    __tablename__ = "si_aggregated"

    si_aggregated_run_id = Column(Integer, ForeignKey("si_aggregated_run.id", ondelete="CASCADE"), primary_key=True)
    python_package_version_entity_id = Column(
        Integer, ForeignKey("python_package_version_entity.id", ondelete="CASCADE"), primary_key=True
    )

    si_aggregated_run = relationship("SecurityIndicatorAggregatedRun", back_populates="python_package_version_entities")
    python_package_version_entity = relationship("PythonPackageVersionEntity", back_populates="si_aggregated_runs")
```

@fridex @saisankargochhayat

Reply:
Added an relation between both tables, please let me know if I am missing something? https://github.com/thoth-station/storages/pull/1963/commits/c6c14a5a8e3cc969df9001a5bcfe9109ba882d8d
Do I have to do an alembic upgrade here?

Reply:
Yes, the migration will need to be created. Also, the syncing logic of SI results needs to be updated so that this relation is created in the database when syncing results.

CC @KPostOffice

Reply:
I don't think we should allow SI workflow to write on PPV directly, only solver should do that for consistency, as many methods depends on PPV, which is equivalent to say that it is solved.
If we introduce this migration and sync part, we are going to have issues, because PPV might be analyzed by SI and not by solver and this would create issues on adviser side as well.

Reply:
mmm actually after looking at when solver sync it creates also entity id (if not existing) and `conditions = [PythonPackageVersion.entity_id == SIAggregated.id]` is considering PPVE from PPV so we do not require PPV in PPVE and no migration. Sorry @saisankargochhayat for the confusion, I must have read too fast the other evening :(

Reply:
Re consistency and issues on adviser side - The precondition for syncing SI workflows is the fact that solvers are already run (we have that already based on the system behavior) so no need to worry about these things.